rts/src/test/java/eta/runtime/io/MemoryManagerTest.java

```diff
@@ -0,0 +1,295 @@
+        copyByteBuffer(b1,5,b2,2,5);
+        b2.position(2).limit(7);
+        b1.position(5).limit(10);
+        assertThat("If you copy some bytes with offset, they should be equal",
+                   b2,is(b1));
+    }
+
+    @Test
+    public void copyByteArrayToItselfNotChangeIt() {
+        ByteBuffer b1=buffer(repeat(10,1));
+        ByteBuffer copy=b1.duplicate();
+        ByteBuffer res=copyByteBuffer(b1,b1,5);
+        assertThat("The returned buffer is the original",
+                   b1,is(sameInstance(res)));
+        assertTrue("The buffer is equal after copy",
+                   b1.clear().equals(copy));
+    }
+
+    @Test
+    public void testCopyByteBufferDups() {
+        ByteBuffer b1=buffer(repeat(10,1)),
+            b2=buffer(repeat(8,2));
+        ByteBuffer cpyB1 = b1.duplicate(),
+            cpyB2=b2.duplicate();
+        ByteBuffer res=copyByteBufferDups(b1,5,b2,2,5);
+        assertThat("The returned buffer is NOT the original",
+                   b1,is(not(sameInstance(res))));
+        assertThat("Position and limit of source does NOT change",
+                   b1,is(cpyB1));
+        assertThat("Position and limit of dest does NOT change",
+                   b2,is(cpyB2));
+        b2.position(2).limit(7);
+        b1.position(5).limit(10);
+        assertThat("If you copy some bytes with offset duplicating buffers, "+
+                   "they should be equal",
+                   b2,is(b1));
+    }
+
+    @Test
+    public void testAllocateBuffer() {
+        int[] sizes = {2,16,256,1024,4096,65536,4194304};
+        for (int size : sizes) {
+            long address = allocateBuffer(size,true);
+            assertTrue("Allocating "+size+" bites: the address must be greater than 0",
+                   address > 0);
+        }
+    } 
+
+    @Test(expected = IllegalArgumentException.class)
+    public void allocateBufferWithNegativeArg() {
+        allocateBuffer(-1,true);
+    }
+    @Test
+    @Ignore
+    public void allocateLargeBuffer() {
+        allocateBuffer(4194305,true);
+    }
+    
+    @Test
+    public void testGetBoundedBuffer() {
+        ByteBuffer b = getBoundedBuffer(0);
+        ByteBuffer empty = ByteBuffer.allocate(0);
+        assertThat("With zero address it returns an empty buffer",
+                   b, is(empty));
+        long addr = allocateBuffer(1024, true);
+        b = getBoundedBuffer(addr);
+        assertTrue("The buffer is as large as the allocated size",
+                   b.remaining() >= 1024);
+        touch(b,(byte)1);
+        ByteBuffer b2 = getBoundedBuffer(addr);
+        assertThat("Two invocations with the same adress ,"+
+                   "should return the same buffer", b2, is(b));
+        b2 = getBoundedBuffer(addr,0,b.remaining());
+        assertThat("Calling it with the same address, "+ 
+                   "without offset and the same length ,"+
+                   "should return the same buffer", b2, is(b));
+    }
+
+    @Ignore
```

Comment:
In MemoryManager.getBoundedBuffer there is a comment that says: 
> Throws an exception  if the block that corresponds to the address has been freed
But this test doesn't throw an exception, we'd should fix the code or the comment

Reply:
Thanks for catching this. I've filed https://github.com/typelead/eta/issues/644 to track it. It's harmless for now but it can make debugging significantly easier if implemented.